flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/HiveCatalog.java

```diff
@@ -226,6 +243,37 @@ public HiveCatalog(
     }
 
     public static HiveConf createHiveConf(
+            @Nullable String hiveConfDir,
+            @Nullable String hadoopConfDir,
+            @Nullable ReadableConfig flinkConfiguration) {
+        HiveConf hiveconf = initHiveConf(hiveConfDir, hadoopConfDir);
+        // add all configuration key with prefix 'flink.hive.hadoop.' and 'flink.hive.' in flink
+        // conf to hive conf
+        String hivePrefix = FLINK_HIVE_CONFIG_PREFIXES;
+        String hadoopPrefix = "hadoop.";
+        if (flinkConfiguration != null) {
+            for (String key : flinkConfiguration.toMap().keySet()) {
+                if (key.startsWith(hivePrefix)) {
+                    String newKey = key.substring(hivePrefix.length());
+                    if (newKey.startsWith(hadoopPrefix)) {
```

Comment:
Why do we need to check starts with hadoopPrefix again? Can we just put to hive conf once it starts with `flink.hive.`?

Reply:
'flink.hive.hadoop.' adds Hadoop configuration for Hive, 
while 'flink.hive.' adds its own configuration to Hive.

like this:
configuration.setString("flink.hive.hadoop.dfs.client.socket-timeout", "5000"); // add hadoop conf for hive
configuration.setString("flink.hive.hive.metastore.client.socket.timeout", "10s"); // add hive conf.

> Why do we need to check starts with hadoopPrefix again? Can we just put to hive conf once it starts with `flink.hive.`?